python_modules/libraries/dagster-fivetran/dagster_fivetran/resources.py

```diff
@@ -608,4 +623,42 @@ def fetch_fivetran_workspace_data(
         Returns:
             FivetranWorkspaceData: A snapshot of the Fivetran workspace's content.
         """
-        raise NotImplementedError()
+        connectors = []
+        destinations = []
+
+        client = self.get_client()
+        groups = client.get_groups()["items"]
+
+        for group in groups:
+            group_id = group["id"]
+
+            destination_details = client.get_destination_details(destination_id=group_id)
+            destinations.append(
+                FivetranContentData(
+                    content_type=FivetranContentType.DESTINATION, properties=destination_details
+                )
+            )
+
+            connectors_details = client.get_connectors_for_group(group_id=group_id)["items"]
+            for connector_details in connectors_details:
+                connector_id = connector_details["id"]
+
+                setup_state = connector_details.get("status", {}).get("setup_state")
+                if setup_state and setup_state in ("incomplete", "broken"):
+                    continue
+
+                schema_config = client.get_schema_config_for_connector(connector_id=connector_id)
+
+                augmented_connector_details = {
+                    **connector_details,
+                    "schema_config": schema_config,
+                    "destination_id": group_id,
+                }
+                connectors.append(
+                    FivetranContentData(
+                        content_type=FivetranContentType.CONNECTOR,
+                        properties=augmented_connector_details,
+                    )
+                )
```

Comment:
> why is the setup_state two gets? Is this actually optional, and what are the cases in which it's optional?

That logic was taken from the legacy code, but the fields are now required in the response, so I will update that.

> I don't like using bare strings to refer to objects ("incomplete", "broken"). Nit; but would prefer a top-level constant.

That's fair, will update.

> I don't feel like the way FivetranContentData is being used here is quite the same as the other integrations (although correct me if I'm wrong)

This is what we do for Power BI, Sigma and Tableau - we augment the data before adding the properties in the content data. That said, I agree that we could wrap these responses in all integrations to keep only what we need and make this easier to maintain. Maybe this could be done separately in another PR as a nice-to-have improvement?

Reply:
I just saw your comment on the next PR. 

For the context, the code here and in the next PR is my mainly based on the legacy code, the goal being to initially reproduce the exact same behavior as we had before but with the new integration pattern. 

I agree that all of this code need improvement and cleaning to make it more maintainable.

Reply:
Yea what I had in mind regarding "other integrations" was `dbt_resource_props = dbt_nodes[unique_id]` from the dbt integration, where we just copy the raw response. Makes sense if this was a property we copied over from the BI integrations, but yea I do think that's kinda an antipattern. If we're making _any_ alterations to the data, the data model should reflect that IMO. 

I'm fine with you doing it in a follow up PR, I understand that rebasing is a pain. That said, I don't necessarily think of it as a nice to have, seeing how it carries over to the translator with all of the additional raw dict indexing we have to do there, I do think it's important to fix.

Reply:
Gotcha, missed that message. Makes sense regarding just switching the pattern first and improving things later.